pkg/dashboards/dashboard.go

```diff
@@ -2,248 +2,114 @@ package dashboards
-	metricTagPackets = "packets"
-)
-
-var allRows []*Row
+type chart struct {
+	metricslatest.Chart
+	mptr *metricslatest.FlowMetric
+}
 
-func init() {
-	for _, scope := range []metricScope{srcDstNodeScope, srcDstNamespaceScope, srcDstWorkloadScope} {
-		// byte/pkt rates
-		for _, valueType := range []string{metricTagBytes, metricTagPackets} {
-			valueTypeText := valueTypeToText(valueType)
-			for _, dir := range []string{metricTagEgress, metricTagIngress} {
-				title := fmt.Sprintf(
-					"%s rate %s %s",
-					valueTypeText,
-					dirToVerb(dir),
-					scope.titlePart,
-				)
-				metric := fmt.Sprintf("%s_%s_%s_total", scope.metricPart, dir, valueType)
-				allRows = append(allRows, row(
-					metric,
-					title,
-					topRatePanels(&scope, metric, scope.joinLabels(), scope.legendPart),
-				))
-			}
-			// drops
-			title := fmt.Sprintf(
-				"%s drop rate %s",
-				valueTypeText,
-				scope.titlePart,
-			)
-			metric := fmt.Sprintf("%s_drop_%s_total", scope.metricPart, valueType)
-			allRows = append(allRows, row(
-				metric,
-				title,
-				topRatePanels(&scope, metric, scope.joinLabels(), scope.legendPart),
-			))
+func createSingleStatPanels(c *chart) []Panel {
+	var panels []Panel
+	for _, q := range c.Queries {
+		title := c.Title
+		if q.Legend != "" {
+			title += ", " + q.Legend
 		}
-		// RTT
-		title := fmt.Sprintf("Round-trip time %s (milliseconds - p99 and p50)", scope.titlePart)
-		metric := fmt.Sprintf("%s_rtt_seconds", scope.metricPart)
-		allRows = append(allRows, row(
-			metric,
-			title,
-			histogramPanels(&scope, metric, scope.joinLabels(), scope.legendPart, "*1000"),
-		))
-		// DNS latency
-		title = fmt.Sprintf("DNS latency %s (milliseconds - p99 and p50)", scope.titlePart)
-		metric = fmt.Sprintf("%s_dns_latency_seconds", scope.metricPart)
-		allRows = append(allRows, row(
-			metric,
-			title,
-			histogramPanels(&scope, metric, scope.joinLabels(), scope.legendPart, "*1000"),
-		))
-		// DNS errors
-		title = fmt.Sprintf("DNS request rate per code and %s", scope.titlePart)
-		metric = fmt.Sprintf("%s_dns_latency_seconds", scope.metricPart)
-		labels := scope.joinLabels() + ",DnsFlagsResponseCode"
-		legend := scope.legendPart + ", {{DnsFlagsResponseCode}}"
-		allRows = append(allRows, row(
-			metric,
-			title,
-			topRatePanels(&scope, metric+"_count", labels, legend),
-		))
+		query := strings.ReplaceAll(q.PromQL, "$METRIC", "netobserv_"+c.mptr.Spec.MetricName)
+		newPanel := NewPanel(title, metricslatest.ChartTypeSingleStat, c.Unit, 3, NewTarget(query, ""))
+		panels = append(panels, newPanel)
 	}
+	return panels
 }
 
-func row(metrics string, title string, panels []Panel) *Row {
-	r := NewRow(title, false, "250px", panels)
-	r.Metric = metrics
-	return r
+func createGraphPanel(c *chart) Panel {
+	var targets []Target
+	for _, q := range c.Queries {
+		query := strings.ReplaceAll(q.PromQL, "$METRIC", "netobserv_"+c.mptr.Spec.MetricName)
+		query = fmt.Sprintf("topk(7, %s)", query)
```

Comment:
100% subjective yes, but from personal testing it was the best compromise to see relevant info without overloading the charts. Top 10 easily leads to overcrowded charts (especially as it's a top per datapoint, not a "global" top); while too small top leads to showing more partial metrics/chunks.
But yes I agree it can be made configurable.
We just need to care about finding a good balance between too much and too few settings.
I guess it doesn't hurt to add that one

Reply:
cool ! thanks @jotak